draft-ietf-oauth-cross-device-security.md

```diff
@@ -666,9 +666,47 @@ As is the case with other kinds of protocols, it can be easy to overlook vulnera
 
-There are various different approaches to formal security analysis and each brings its own strengths and weaknesses. For example, models differ in the level of detail in which they can capture a protocol (granularity, expressiveness), in the kind of statements they can produce, and whether the proofs can be assisted by tools or have to be performed manually. One of the most successfully used approaches is the so-called Web Infrastructure Model (WIM), a model specifically designed for the analysis of web authentication and authorization protocols. While it is a manual (pen-and-paper) model, it captures details of browsers and web interactions in unprecedented detail. Using the WIM, previously unknown flaws in OAuth, OpenID Connect, and FAPI were discovered.
+There are various different approaches to formal security analysis and each brings its own strengths and weaknesses. For example, models differ in the level of detail in which they can capture a protocol (granularity, expressiveness), in the kind of statements they can produce, and whether the proofs can be assisted by tools or have to be performed manually.
+
+The following works have been identified as relevant to the analysis of cross-device flows:
+
+ * In "Formal analysis of self-issued OpenID providers" [@Bauer2022], the
+   protocol of [@OpenID.SIOPV2] was analyzed using the Web Infrastructure Model
+   (WIM). The WIM is specifically designed for the analysis of web
+   authentication and authorization protocols. While it is a manual
+   (pen-and-paper) model, it captures details of browsers and web interactions
+   to a degree that is hard to match in automated models. In previous works,
+   previously unknown flaws in OAuth, OpenID Connect, and FAPI were discovered
+   using the WIM. In the analysis of a cross-device SIOP V2 flow in
+   [@Bauer2022], the request replay attack already described in Section 13.3 of
+   [@OpenID.SIOPV2] was confirmed in the model. A mitigation was implemented
+   based on a so-called Cross-Device Stub, essentially a component that serves
+   to link the two devices before the protocol flow starts. This essentially
+   creates a trusted device relationship as described in (#trusted_devices). The
+   mitigation was shown to be effective in the model.
+ * In "Security analysis of the Grant Negotiation and Authorization Protocol"
+   [@Helmschmidt2022], an analysis of a draft of the Grant Negotiation and
+   Authorization Protocol (GNAP) [@I-D.ietf-gnap-core-protocol] was performed
+   using the Web Infrastructure Model. The same attack as in [@Bauer2022] was
+   found to apply to GNAP as well. In this case, a model of a "careful user"
+   (see (#user_education)) was used to show that the attack can be
+   prevented (at least in theory) by the user.
+ * In "The Good, the Bad and the (Not So) Ugly of Out-of-Band Authentication
```

Comment:
* In "The Good, the Bad and the (Not So) Ugly of Out-of-Band Authentication with eID Cards and Push Notifications: Design, Formal and Risk Analysis" [@Pernpruner2020], Pernpruner et al. formally analysed an authentication protocol relying on push notifications delivered to an out-of-band device to approve the authentication attempt on the primary device (Backchannel-Transferred Session Pattern, (#btsp)). The analysis was performed using the specification language ASLan++ and the model checker SATMC. According to the results of the analysis, they identified and defined the category of *implicit attacks*, which are those managing to deceive users into approving a malicious authentication attempt through social engineering techniques, thus not compromising all the authentication factors involved; these attacks are aligned with the definition of Cross-Device Consent Phishing (CDCP) attacks. (TODO: Add references to mitigations once included in the draft.)